src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const result = [];
+
+  for (const type of actions) {
+    switch (type.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, type.extraData);
+
+        result.push({ ...stateCopy });
+        break;
+      case 'removeProperties':
+        for (const key of type.keysToRemove) {
+          delete stateCopy[key];
+        }
+
+        result.push({ ...stateCopy });
+        break;
+      case 'clear':
+        Object.keys(stateCopy).forEach(key => (delete stateCopy[key]));
+
+        result.push({ ...stateCopy });
+        break;
+    }
```

Comment:
[CODE STYLE]: switch/case should always have default case for error handling.

Add default